src/lib.rs

```
@@ -0,0 +1,100 @@
+//! Pull request management for bare repos
+use std::io;
+use std::ffi::OsString;
+use std::path::Path;
+use std::process::Command;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+#[derive(Debug)]
+pub struct Git {
+    program: OsString,
+}
+
+impl Git {
+    pub fn new() -> Git {
+        Git{ program: OsString::from("git") }
+    }
+    pub fn with_path(path: &Path) -> Git {
+        Git{ program: path.as_os_str().to_os_string() }
+    }
+    pub fn version(&self) -> io::Result<String> {
+        let output = Command::new(&self.program).arg("--version").output()?;
+        if !output.status.success() {
+            return Err(io::Error::new(io::ErrorKind::Other, "git --version"));
```

The actual error is lost here. Create a custom struct that includes exit status and anything else we need.

---

Need to update all functions in this struct, not just this one.

---

This is fixed in 5f90c9f